### PR TITLE
Fix simulator docker: CPU only for now

### LINKED ISSUE #2674: Display "Planner Solution Error" When run simulation with last openpilot docker image
**Describe the bug**

Display "Planner Solution Error" When run simulation with last openpilot docker image

**How to reproduce or log data**

1../start_carla.sh
2./start_openpilot_docker.sh
3.comment out related line in issue 
https://github.com/commaai/openpilot/issues/2501
4.press "1" in bridge.py terminal

**Expected behavior**

simulation normally

**Additional context**

NA

Operating system: Ubuntu 20.04
2020-12-02 21-16-57屏幕截图

### LINKED ISSUE #19608: onnxmodel pwrite broken pipe on CUDAExecutionProvider
**Describe the bug**

When starting OP on Ubuntu for simulation, sometimes the following error would occur and modeld won't work. This is likely caused by a race condition somewhere since it works fine sometimes.

> _modeld: selfdrive/modeld/runners/onnxmodel.cc:63: void ONNXModel::pwrite(float *, int): Assertion `err >= 0' failed.

The specific reason for this error is writing to a broken pipe according to the `errno`, but I don't know why exactly this happens.

**How to reproduce or log data**

Run OP on Ubuntu with CUDAExecutionProvider

**Expected behavior**

modeld works properly

**Additional context**

I tested for this bug multiple times, from what I'm seeing so far, it rarely happens on onnx's normal CPUExecutionProvider, and once I switch to the faster CUDAExecutionProvider, this problem begins to happen more frequently. This could be the race condition itself, or maybe it's just my testing environment.

Operating system: Ubuntu 20.10

### LINKED ISSUE #20060: modeld and sensord failing to start  in PC simulation
I'm trying to make simulator running in UBUNTU 20.04 but it's failing to start modeld and sensord with following message:

selfdrive/sensord/sensors/bmx055_accel.cc: Reading chip ID failed: -1
selfdrive/sensord/sensors_qcom2.cc: Error initializing sensors
selfdrive/loggerd/loggerd.cc: logging to /home/palash/.comma/media/0/realdata/2021-02-11--00-43-35--0
platform[0] CL_PLATFORM_NAME: NVIDIA CUDA
vendor: NVIDIA Corporation
platform version: OpenCL 1.2 CUDA 11.0.228
profile: FULL_PROFILE
extensions: cl_khr_global_int32_base_atomics cl_khr_global_int32_extended_atomics cl_khr_local_int32_base_atomics cl_khr_local_int32_extended_atomics cl_khr_fp64 cl_khr_byte_addressable_store cl_khr_icd cl_khr_gl_sharing cl_nv_compiler_options cl_nv_device_attribute_query cl_nv_pragma_unroll cl_nv_copy_opts cl_nv_create_buffer cl_khr_int64_base_atomics cl_khr_int64_extended_atomics
name :GeForce RTX 2080
device version :OpenCL 1.2 CUDA
max work group size :1024
type = 4 = CL_DEVICE_TYPE_GPU
platform[0] CL_PLATFORM_NAME: NVIDIA CUDA
vendor: NVIDIA Corporation
platform version: OpenCL 1.2 CUDA 11.0.228
profile: FULL_PROFILE
extensions: cl_khr_global_int32_base_atomics cl_khr_global_int32_extended_atomics cl_khr_local_int32_base_atomics cl_khr_local_int32_extended_atomics cl_khr_fp64 cl_khr_byte_addressable_store cl_khr_icd cl_khr_gl_sharing cl_nv_compiler_options cl_nv_device_attribute_query cl_nv_pragma_unroll cl_nv_copy_opts cl_nv_create_buffer cl_khr_int64_base_atomics cl_khr_int64_extended_atomics
name :GeForce RTX 2080
device version :OpenCL 1.2 CUDA
max work group size :1024
type = 4 = CL_DEVICE_TYPE_GPU
modelV2: Reader was evicted, reconnecting
cameraOdometry: Reader was evicted, reconnecting
vin query retry (1) ...
selfdrive/modeld/modeld.cc: models loaded, modeld starting
vin query retry (2) ...
Assertion failed: ok (src/mailbox.cpp:99)
cameraOdometry: Reader was evicted, reconnecting
width 1164, height 874, rgb_stride 3492
vin query retry (3) ...
width 1632, height 1224, rgb_stride 4896
Starting listener for: camerad
visionipc_camerad_0: Reader was evicted, reconnecting
pandad thermald logmessaged ui uploader deleter controlsd plannerd loggerd radard calibrationd paramsd camerad modeld proclogd locationd clocksd logcatd sensord
vin query retry (4) ...
visionipc_camerad_3: Reader was evicted, reconnecting
selfdrive/loggerd/loggerd.cc: camera 0 rotate encoder to /home/palash/.comma/media/0/realdata/2021-02-11--00-43-35--0
selfdrive/modeld/modeld.cc: connected with buffer size: 1526004 (1164 x 874)
visionipc_camerad_3: Reader was evicted, reconnecting
vin query retry (5) ...
_modeld: selfdrive/modeld/runners/onnxmodel.cc:63: void ONNXModel::pwrite(float *, int): Assertion `err >= 0' failed.

Any insight what is going wrong here?

### PR DESCRIPTION
**Pull request by iejMac**
fixes #20060, fixes #19608, fixes #2674

- [x] building in docker
- [x] cleanup/simplify dockerfile
- [x] engages with no errors in sim

### CODE REVIEW

File: selfdrive/manager.py [hunk 1]
```diff
@@ -253,10 +253,10 @@ def get_running():
...
     'rtshield',
   ]
-else:
-   car_started_processes += [
-    'sensord',
-  ]
+# else:
+     #car_started_processes += [
+    # 'sensord',
+  # ]
```
Comment (adeebshihadeh) [inline selfdrive/manager.py:259]:
```suggestion
elif not PC:
  car_started_processes += [
    'sensord',
  ]
```

File: selfdrive/manager.py [hunk 2]
```diff
@@ -214,7 +214,7 @@ def get_running():
 
 if EON:
   persistent_processes += [
-    'sensord',
+    # 'sensord',
```
Comment (adeebshihadeh) [inline selfdrive/manager.py:217]: revert

File: selfdrive/manager.py [hunk 3]
```diff
@@ -171,7 +171,7 @@ def build():
   "calibrationd": "selfdrive.locationd.calibrationd",
   "paramsd": "selfdrive.locationd.paramsd",
   "camerad": ("selfdrive/camerad", ["./camerad"]),
-  "sensord": ("selfdrive/sensord", ["./sensord"]),
+  # "sensord": ("selfdrive/sensord", ["./sensord"]),
```
Comment (adeebshihadeh) [inline selfdrive/manager.py:174]: revert

File: tools/replay/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+Replay driving data
```
Comment (adeebshihadeh) [inline tools/replay/README.md:1]: revert this file

File: tools/sim/bridge.py [hunk 5]
```diff
@@ -278,7 +277,7 @@ def destroy():
       sm.update(0)
       throttle_op = sm['carControl'].actuators.gas #[0,1]
       brake_op = sm['carControl'].actuators.brake #[0,1]
-      steer_op = sm['controlsState'].angleSteersDes # degrees [-180,180]
+      steer_op = sm['controlsState'].steeringAngleDesiredDeg # angleSteersDes # degrees [-180,180]
```
Comment (adeebshihadeh) [inline tools/sim/bridge.py:280]: fix the comment

File: selfdrive/modeld/runners/onnx_runner.py [hunk 6]
```diff
@@ -1,10 +1,15 @@
 #!/usr/bin/env python3
-# TODO: why are the keras models saved with python 2?
-from __future__ import print_function
 
 import os
 import sys
 import numpy as np
+
+os.environ["OMP_NUM_THREADS"] = "1"
```
Comment (adeebshihadeh) [inline selfdrive/modeld/runners/onnx_runner.py:7]: don't think we need all of these

File: selfdrive/modeld/modeld.cc [hunk 7]
```diff
@@ -118,7 +118,7 @@ int main(int argc, char **argv) {
   model_init(&model, device_id, context);
   LOGW("models loaded, modeld starting");
 
-  VisionIpcClient vipc_client = VisionIpcClient("camerad", VISION_STREAM_YUV_BACK, true, device_id, context);
+  VisionIpcClient vipc_client = VisionIpcClient("camerad", VISION_STREAM_RGB_BACK, true, device_id, context);
```
Comment (adeebshihadeh) [inline selfdrive/modeld/modeld.cc:121]: revert this

File: selfdrive/manager.py [hunk 8]
```diff
@@ -253,8 +253,8 @@ def get_running():
   car_started_processes += [
     'rtshield',
   ]
-else:
-   car_started_processes += [
+elif not PC:
+  car_started_processes += [
```
Comment (adeebshihadeh) [inline selfdrive/manager.py:257]:
```suggestion
   car_started_processes += [
```

File: selfdrive/debug/capnp2json.py [hunk 9]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (adeebshihadeh) [inline selfdrive/debug/capnp2json.py:1]: looks like this was committed by mistake

File: tools/sim/Dockerfile.sim [hunk 10]
```diff
@@ -6,40 +6,32 @@ RUN apt-get update && apt-get install -y \
...
-ENV NVIDIA_DRIVER_CAPABILITIES ${NVIDIA_DRIVER_CAPABILITIES},display
-RUN apt-get update && apt-get install -y --no-install-recommends \
-      mesa-utils \
-      ocl-icd-libopencl1 \
-    && rm -rf /var/lib/apt/lists/*
-RUN mkdir -p /etc/OpenCL/vendors && \
-    echo "libnvidia-opencl.so.1" > /etc/OpenCL/vendors/nvidia.icd
 ENV NVIDIA_VISIBLE_DEVICES all
-ENV NVIDIA_DRIVER_CAPABILITIES compute,utility,display
+ENV NVIDIA_DRIVER_CAPABILITIES graphics,utility,compute
```
Comment (adeebshihadeh) [inline tools/sim/Dockerfile.sim:34]: does this do anything now that we removed the gpu driver?

File: tools/sim/Dockerfile.sim [hunk 11]
```diff
@@ -6,40 +6,32 @@ RUN apt-get update && apt-get install -y \
...
-    echo /opt/intel/*/lib64/libintelocl.so > /etc/OpenCL/vendors/intel.icd; \
+RUN echo INTEL_DRIVER is $INTEL_DRIVER && \
+    curl -O $INTEL_DRIVER_URL/$INTEL_DRIVER && \
+    tar -xzf $INTEL_DRIVER && \
+    for i in $(basename $INTEL_DRIVER .tgz)/rpm/*.rpm; do alien --to-deb $i; done && \
+    dpkg -i *.deb && \
+    rm -rf $INTEL_DRIVER $(basename $INTEL_DRIVER .tgz) *.deb && \
+    mkdir -p /etc/OpenCL/vendors && \
+    echo /opt/intel/opencl_compilers_and_libraries_18.1.0.015/linux/compiler/lib/intel64_lin/libintelocl.so > /etc/OpenCL/vendors/intel.icd && \
     rm -rf /tmp/opencl-driver-intel;
```
Comment (adeebshihadeh) [inline tools/sim/Dockerfile.sim:31]: remove trailing semicolon

File: selfdrive/ui/ui [hunk 12]
```diff
@@ -1,3 +1,3 @@
 #!/bin/sh
 export LD_LIBRARY_PATH="/system/lib64:$LD_LIBRARY_PATH"
-exec ./_ui
+exec ./_ui --no-sandbox
```
Comment (adeebshihadeh) [inline selfdrive/ui/ui:3]: just set `QTWEBENGINE_DISABLE_SANDBOX` in the dockerfile for now

File: tools/sim/start_openpilot_docker.sh [hunk 13]
```diff
@@ -1,17 +1,17 @@
 #!/bin/bash
 
 # expose X to the container
-xhost +local:root
-docker pull commaai/openpilot-sim:latest
+sudo xhost +local:root
```
Comment (adeebshihadeh) [inline tools/sim/start_openpilot_docker.sh:4]: are you sure this needs sudo? how did it work before?

File: tools/sim/start_openpilot_docker.sh [hunk 14]
```diff
@@ -1,17 +1,17 @@
...
   --gpus all \
-  --device=/dev/dri/  \
+  --device=/dev/dri \
   -v /tmp/.X11-unix:/tmp/.X11-unix \
   --shm-size 1G \
   -e DISPLAY=$DISPLAY \
+  -e QT_X11_NO_MITSHM=1 \
   commaai/openpilot-sim:latest \
-  /bin/bash -c "cd tools && cd sim && sh tmux_script.sh $*"
+  /bin/bash -c "cd /openpilot/tools && cd sim && sh tmux_script.sh $*"
```
Comment (adeebshihadeh) [inline tools/sim/start_openpilot_docker.sh:17]:
```suggestion
  /bin/bash -c "cd /openpilot/tools/sim && ./tmux_script.sh $*"
```

File: tools/sim/Dockerfile.sim (hunk 10 above)
Comment (iejMac) [reply tools/sim/Dockerfile.sim:34]: Yes. UI is black if this is commented out

**Pull request merged**